controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -1041,6 +1048,33 @@ func (r *ReconcileHyperConverged) migrateBeforeUpgrade(req *common.HcoRequest) (
 	return upgradePatched, nil
 }
 
+func (r *ReconcileHyperConverged) removeOldApifromCrd(req *common.HcoRequest, crdName, removedApiVersion string) error {
+	var crd apiextensionsv1.CustomResourceDefinition
+	if err := r.client.Get(req.Ctx, types.NamespacedName{Name: crdName}, &crd); err != nil {
+		req.Logger.Error(err, fmt.Sprintf("unable to get %s CRD", crdName))
+		return err
+	}
+
+	if !slices.Contains(crd.Status.StoredVersions, removedApiVersion) {
+		return nil
+	}
+	var newStoredVersions []string
+	for _, ver := range crd.Status.StoredVersions {
+		if ver == removedApiVersion {
+			continue
+		}
+		newStoredVersions = append(newStoredVersions, ver)
+	}
```

Comment:
I think we can use methods from the slices package, instead of this loop.; e.g.
```suggestion
	i := slices.Index(crd.Status.StoredVersions, removedApiVersion)
	if i == -1 {
		return nil
	}
	
	slices.Delete(crd.Status.StoredVersions, i, i+1)
```

I think it's cleaner. WDYT?

Reply:
But actually, I think we should try to use json patch here. So maybe the we only need `slices.Index`: if it `-1`, we can we're fine. if not, we can build a json patch based on this index.